marshal/unmarshal.go

```diff
@@ -60,26 +60,24 @@ func unmarshalInternal(data []byte, v any) error {
 	}
 
 	value := reflect.ValueOf(v)
-	derefer := reflect.Indirect(value)
+	value = reflect.Indirect(value)
 
-	objectType := derefer.Type()
+	objectType := value.Type()
 
 	foundFields := []string{}
-	fields := getTypeFields(objectType)
 
-	for i := 0; i < derefer.NumField(); i++ {
-		field := derefer.Field(i)
-		fieldType := field.Type()
-		jsonTag := fields[i].JsonTag
-
-		if fields[i].Anonymous {
-			setField(field, data)
+	for _, field := range getTypeFields(objectType) {
+		if field.JsonTag.ignore {
```

Comment:
Can't we opt to not include the type in the cached type slice to begin with, if it's ignored anyway?

Reply:
We need it when searching for the ForceSendField name (which is ignored, but uses the same cache)